verilator-config.cmake.in

```diff
@@ -0,0 +1,383 @@
+#
+# Include it in your CMakeLists.txt using:
+#
+#     find_package(verilate)
+#
+#  This script adds a verilate function.
+#
+#     add_executable(simulator <your-c-sources>)
+#     verilate(simulator SOURCES <your-hdl-sources>)
+#
+# Copyright 2003-2019 by Wilson Snyder. This program is free software; you can
+# redistribute it and/or modify it under the terms of either the GNU
+# Lesser General Public License Version 3 or the Perl Artistic License
+# Version 2.0.
+#
+######################################################################
+
+cmake_minimum_required(VERSION 3.8)
+
+# Prefer VERILATOR_ROOT from environment
+if (DEFINED ENV{VERILATOR_ROOT})
+  set(VERILATOR_ROOT "$ENV{VERILATOR_ROOT}" CACHE PATH "VERILATOR_ROOT")
+endif()
+
+set(VERILATOR_ROOT "${CMAKE_CURRENT_LIST_DIR}" CACHE PATH "VERILATOR_ROOT")
+
+find_program(VERILATOR_BIN NAMES verilator_bin verilator_bin.exe
+  HINTS ${VERILATOR_ROOT}/bin ENV VERILATOR_ROOT
+  NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH)
+
+if (NOT VERILATOR_ROOT)
+  message(FATAL_ERROR "VERILATOR_ROOT cannot be detected. Set it to the appropriate directory (e.g. /usr/share/verilator) as an environment variable or CMake define.")
+endif()
+
+if (NOT VERILATOR_BIN)
+  message(FATAL_ERROR "Cannot find verilator_bin excecutable.")
+endif()
+
+set(verilator_FOUND 1)
+
+include(CheckCXXSourceCompiles)
+function(_verilator_check_cxx_libraries LIBRARIES RESVAR)
+  # Check whether a particular link option creates a valid executable
+  set(_VERILATOR_CHECK_CXX_LINK_OPTIONS_SRC "int main() {return 0;}\n")
+  set(CMAKE_REQUIRED_FLAGS)
+  set(CMAKE_REQUIRED_DEFINITIONS)
+  set(CMAKE_REQUIRED_INCLUDES)
+  set(CMAKE_REQUIRED_LINK_OPTIONS)
+  set(CMAKE_REQUIRED_LIBRARIES ${LIBRARIES})
+  set(CMAKE_REQUIRED_QUIET)
+  check_cxx_source_compiles("${_VERILATOR_CHECK_CXX_LINK_OPTIONS_SRC}" "${RESVAR}")
+  set("${RESVAR}" "${${RESVAR}}" PARENT_SCOPE)
+endfunction()
+
+# Check compiler flag support. Skip on MSVC, these are all GCC flags.
+if (NOT CMAKE_CXX_COMPILER_ID MATCHES MSVC)
+  if (NOT DEFINED VERILATOR_CFLAGS OR NOT DEFINED VERILATOR_MT_CFLAGS)
+    include(CheckCXXCompilerFlag)
+    foreach (FLAG @CFG_CXX_FLAGS_CMAKE@)
+      string(MAKE_C_IDENTIFIER ${FLAG} FLAGNAME)
+      check_cxx_compiler_flag(${FLAG} ${FLAGNAME})
+      if (${FLAGNAME})
+        list(APPEND VERILATOR_CFLAGS ${FLAG})
+      endif()
+    endforeach()
+    foreach (FLAG @CFG_LDFLAGS_THREADS_CMAKE@)
+      string(MAKE_C_IDENTIFIER ${FLAG} FLAGNAME)
+      _verilator_check_cxx_libraries("${FLAG}" ${FLAGNAME})
+      if (${FLAGNAME})
+        list(APPEND VERILATOR_MT_CFLAGS ${FLAG})
+      endif()
+    endforeach()
+  endif()
+endif()
+
+define_property(TARGET
+  PROPERTY VERILATOR
+  BRIEF_DOCS "Verilator"
+  FULL_DOCS "Verilator"
+)
+
+define_property(TARGET
+  PROPERTY VERILATOR_PREFIXES
+  BRIEF_DOCS "List of Verilated prefixes added to this target"
+  FULL_DOCS "List of Verilated prefixes added to this target"
+)
```

Comment:
@madebr `VERILATOR` and `VERILATOR_PREFIXES` seem unnecessary, do you need the for something? `VERILATOR_PREFIXES` doesn't seem to be used anywhere, and we could just always do `target_link_libraries(${TARGET} PUBLIC ${VERILATOR_MT_CFLAGS})`

Reply:
Yup, they are not really needed per se.
They are more of a hint to users that this targets contains VERILATED modules.